tests/security/test_python_code/test_python_flaws.py

```diff
@@ -0,0 +1,224 @@
+    b_mgr.run_tests()
+
+    # Trigger output of results by Bandit Manager
+    _, filename = tempfile.mkstemp(suffix='.json')
+    b_mgr.output_results(None,  # context lines
+                         severity_level,  # minimum severity level
+                         confidence_level,  # minimum confidence level
+                         open(filename, mode='w'),  # output file object
+                         'json',  # output format
+                         None)  # msg template
+
+    # Read the temporary file with the Bandit result and remove it
+    with open(filename, mode="r") as f:
+        bandit_output = json.load(f)
+    os.remove(filename)
+
+    return bandit_output
+
+
+def update_known_flaws(known_flaws: dict, results: list) -> dict:
+    """Compare the Bandit results of the run with the already known flaws. Update the known flaws with the new line
+    numbers and remove the known flaws that don't appear in the run results (were fixed).
+
+    Args:
+        known_flaws (dict): Dictionary containing already known vulnerabilities or false positives detected by Bandit.
+        results (list): List containing all the possible flaws obtained in the Bandit run.
+
+    Returns:
+        dict: Updated known flaws.
+    """
+    updated_known_flaws = {k: None for k in known_flaws.keys()}
+    for key in known_flaws.keys():
+        for i in range(len(known_flaws[key])):
+            next_flaw = next((flaw for flaw in results
+                              if (flaw['code'] == known_flaws[key][i]['code']
+                                  and flaw['filename'] == known_flaws[key][i]['filename']
+                                  and flaw['test_id'] == known_flaws[key][i]['test_id'])), {})
+            if next_flaw:
+                known_flaws[key][i] = next_flaw
+            else:
+                known_flaws[key][i] = None
+        updated_known_flaws[key] = [flaw for flaw in known_flaws[key] if flaw]
+
+    return updated_known_flaws
+
+
+@pytest.fixture(scope='session', autouse=True)
+def clone_wazuh_repository(pytestconfig):
+    """Fixture that clones a Wazuh repository in a temporary directory and checkout to the branch given by parameter.
+    Remove the temporary directory once the test session using this fixture has finished.
+
+    Args:
+        pytestconfig (fixture): Session-scoped fixture that returns the :class:`_pytest.config.Config` object.
+
+    Yields:
+        Union[str, None]: The temporary directory name or None if the clone or checkout actions were not successful.
+    """
+    # Get Wazuh repository and branch
+    repository_name = pytestconfig.getoption('repo')
+    branch = pytestconfig.getoption('branch')
+
+    # Create temporary dir
+    t = tempfile.mkdtemp()
+
+    try:
+        # Clone into temporary dir
+        repo = Repo.clone_from(f"https://github.com/wazuh/{repository_name}.git", t)
+        # Checkout to given branch
+        repo.git.checkout(branch)
```

Comment:
We need to investigate if we could add the option --branch and --depth=1 in order to make this faster